process/setup_sp.py

```diff
@@ -373,6 +373,100 @@ def process_distant_nodes(
     return(distant_nodes)
 
 
+#Cumulative opportunities (binary)
+#1 if d <= access_dist
+#0 if d > access_dist
+def binary_access_score(df, new_name, old_name, access_dist):
+    """
+    Calculate accessibiity score using binary measure: 1 if access <= access_dist, 0 otherwise
+
+    Parameters
+    ----------
+    df: DataFrame
+        DataFrame with origin-destination distances
+    new_name: list
+        list of new field names for new access score
+    old_name: list
+         list of original field names for distances
+    access_dist: int
+        access distance threshold
+
+    Returns
+    -------
+    DataFrame
+    """
+    df[new_name] = (df[old_name] <= access_dist).astype("Int64").fillna(0)
+    return df[new_name]
+
+
+#Soft threshold access score
+#Higgs, C., Badland, H., Simons, K. et al. The Urban Liveability Index
+def soft_access_score(df, new_name, old_name, access_dist, k):

```

Comment:
Two changes I would make to these functions are
1) No need to provide the 'new name' as this is defined externally to the function, we just need to return a dataframe with some values
2) It would be good to provide default values (if possibly based on reference) to suggest sensible 'k' given 'access_dist'

So, this function could be re-written as
```
#Soft threshold access score
#Higgs, C., Badland, H., Simons, K. et al. The Urban Liveability Index
def soft_access_score(df, distances, access_dist=500, k=5):
    """
    Calculate accessibiity score using soft threshold approach:
    1 / (1+ e ^(k *((dist-access_dist)/access_dist)))
    Parameters
    ----------
    df: DataFrame
        DataFrame with origin-destination distances
    distances: list
        list of new field names for new access score columns
    access_dist: int
        access distance threshold
    k: int
        the slope of decay
    Returns
    -------
    DataFrame
    """
    scores = (1 / (1+numpy.exp(k * ((df[distances]-access_dist) / access_dist)))).astype(float).fillna(0)
    return(scores)
```

An example run through of this shows it appears to work --- (I think this removal of 'old_name' will make the cumulative Gaussian function in particular simpler, avoiding the loop)

```
df = pandas.DataFrame({'x1':[100,200,300,400,500,600,700,800,900,1000],'x2':[1000,900,800,700,600,500,400,300,200,100]})

distances = ['x1','x2'] 
access_dist = 500
k = 5

df[[f'soft_{x}' for x in distances]] =  soft_access_score(df, distances, access_dist=500, k=5)
df
```
and this gives
```
     x1    x2   soft_x1   soft_x2
0   100  1000  0.982014  0.006693
1   200   900  0.952574  0.017986
2   300   800  0.880797  0.047426
3   400   700  0.731059  0.119203
4   500   600  0.500000  0.268941
5   600   500  0.268941  0.500000
6   700   400  0.119203  0.731059
7   800   300  0.047426  0.880797
8   900   200  0.017986  0.952574
9  1000   100  0.006693  0.982014
```

Reply:
Thanks, Carl. This seems more straightforward. I will make the two changes you proposed.